app/frontend/wallet/account-manager.ts

```diff
@@ -0,0 +1,65 @@
+import NamedError from '../helpers/NamedError'
+import {Account} from './account'
+import {MAX_ACCOUNT_COUNT} from './constants'
+
+const AccountManager = ({config, cryptoProvider, blockchainExplorer}) => {
+  const accounts: Array<ReturnType<typeof Account>> = []
+
+  function getAccount(accountIndex: number) {
+    return accounts[accountIndex]
+  }
+
+  function discoverNewAccount() {
```

Comment:
discoverNextAccount seems like a more suitable name. The "new" keyword does not really reflect that the discovery is performed in a sequential order

Reply:
I agree `next` is slightly better :+1: